docs/nightly/en/reference/sql/data-types.md

```diff
@@ -64,6 +64,126 @@ Output:
+
+Let's take some examples:
+
+```sql
+SELECT '2 years 15 months 100 weeks 99 hours 123456789 milliseconds'::INTERVAL;
+```
+
+```sql
++---------------------------------------------------------------------+
+| Utf8("2 years 15 months 100 weeks 99 hours 123456789 milliseconds") |
++---------------------------------------------------------------------+
+| P3Y3M700DT133H17M36.789S                                            |
++---------------------------------------------------------------------+
+```
+
+55 minutes ago:
+
+```sql
+SELECT '-1 hour 5 minute'::INTERVAL;
+```
+
+```sql
++--------------------------+
+| Utf8("-1 hour 5 minute") |
++--------------------------+
+| P0Y0M0DT0H-55M0S         |
++--------------------------+
+```
+
+1 hour and 5 minutes ago:
+
+```sql
+SELECT '-1 hour -5 minute'::INTERVAL;
+```
+
+```sql
++---------------------------+
+| Utf8("-1 hour -5 minute") |
++---------------------------+
+| P0Y0M0DT-1H-5M0S          |
++---------------------------+
+```
+
+And of course, you can manipulate time with intervals by arithmetics.
+Get the time of 5 minutes go:
+
+```sql
+SELECT now() - INTERVAL '5 minute';
+```
+
+```sql
++----------------------------------------------+
+| now() - IntervalMonthDayNano("300000000000") |
++----------------------------------------------+
+| 2024-06-24 21:24:05.012306                   |
++----------------------------------------------+
+```
+
+Note that you can also input the interval type using the `INTERVAL 'literal'` format. Using the syntax `'-1 hour -5 minute'::INTERVAL` explicitly casts the string to an interval type, which is how SQL handles type conversion.
+
+GreptimeDB also supports shorthand forms without spaces, such as `3y2mon4h`, but they must be written in the `INTERVAL 'literal'` format:
+
+```sql
+SELECT INTERVAL '3y2mon4h';
+```
+
+```sql
++---------------------------------------------------------+
+| IntervalMonthDayNano("3010670175542044842954670112768") |
++---------------------------------------------------------+
+| P3Y2M0DT4H0M0S                                          |
++---------------------------------------------------------+
+```
+
+It also supports signed numbers:
+
+```sql
+SELECT INTERVAL '-1h5m';
+```
+
+```sql
++----------------------------------------------+
+| IntervalMonthDayNano("18446740773709551616") |
++----------------------------------------------+
+| P0Y0M0DT0H-55M0S                             |
++----------------------------------------------+
+```
+
+Supported abbreviations include:
+
+| Abbreviation | Full name |
+|-------|---------------|
+| y     | years         |
+| mon   | months        |
+| w     | weeks         |
+| d     | days          |
+| h     | hours         |
+| m     | minutes       |
+| s     | seconds       |
+| millis| milliseconds  |
+| mils  | milliseconds          |
+| ms    | milliseconds  |
```

Comment:
based on [expand_interval.rs](https://github.com/GreptimeTeam/greptimedb/blob/948c8695d0ec222a04cc2520a56d4f066927a9b0/src/sql/src/statements/transform/expand_interval.rs#L39)  it should be microseconds.
But I'm not sure if I pick the correct `Abbreviation`. Either code should be adjusted or docs.

Reply:
@etolbakov I think `ms` is mostly used for milliseconds.

Reply:
Let's adjust the code.

Reply:
Is there any source to use `mils` as an abbr.? I know `millis` and `ms` but have never seen `mils`.

Reply:
happy to drop `mils`, I've made up this abbreviation.

Reply:
Let's merge this pull request. We can create a new one to update when your fix is ready. @etolbakov Thank you.

Reply:
Sounds good 👍